app/models/champs/dgfip_champ.rb

```diff
@@ -0,0 +1,53 @@
+# == Schema Information
+#
+# Table name: champs
+#
+#  id                             :integer          not null, primary key
+#  data                           :jsonb
+#  fetch_external_data_exceptions :string           is an Array
+#  private                        :boolean          default(FALSE), not null
+#  rebased_at                     :datetime
+#  row                            :integer
+#  type                           :string
+#  value                          :string
+#  value_json                     :jsonb
+#  created_at                     :datetime
+#  updated_at                     :datetime
+#  dossier_id                     :integer
+#  etablissement_id               :integer
+#  external_id                    :string
+#  parent_id                      :bigint
+#  type_de_champ_id               :integer
+#
+class Champs::DgfipChamp < Champs::TextChamp
+  # see https://github.com/betagouv/api-particulier/blob/master/src/presentation/middlewares/dgfip-input-validation.middleware.ts
+  validates :numero_fiscal, format: { with: /\A\w{13,14}\z/ }, if: -> { reference_avis.present? && validation_context != :brouillon }
+  validates :reference_avis, format: { with: /\A\w{13,14}\z/ }, if: -> { numero_fiscal.present? && validation_context != :brouillon }
+
+  store_accessor :value_json, :numero_fiscal, :reference_avis
+
+  def blank?
+    external_id.nil?
+  end
+
+  def fetch_external_data?
+    true
+  end
+
+  def fetch_external_data
+    return if !valid?
+
+    APIParticulier::DgfipAdapter.new(
+      procedure.api_particulier_token,
+      numero_fiscal,
+      reference_avis,
+      procedure.api_particulier_sources
+    ).to_params
+  end
```

Comment:
Si on veut coller au champ cnaf

```suggestion
  def fetch_external_data
    if valid?
    APIParticulier::DgfipAdapter.new(
      procedure.api_particulier_token,
      numero_fiscal,
      reference_avis,
      procedure.api_particulier_sources
    ).to_params
  end
```

Reply:
Là aussi, petit habitude personnelle : j'essaie généralement retourner une valeur au plus tôt et d'éviter les cascades d'instructions.